orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/TemplatedPipelineRequest.java

```diff
@@ -27,7 +26,7 @@
   String id;
   String schema;
   String type;
-  List<ExpectedArtifact> expectedArtifacts;
+  List<HashMap<String, Object>> expectedArtifacts;
```

Comment:
The only reason to move it to `HashMap<String, Object>` is that it's easier to merge artifacts from different sources given that the other sources are using `HashMap<String, Object>` (`TemplateConfiguration.PipelineConfiguration` and `PipelineTemplate.Configuration`).

I considered changing these 2 configuration classes to use `ExpectedArtifact` instead but many tests failed because of the lack of `Artifact boundArtifact` field in templates and configurations.

Given that orca is not using the expected artifacts in this class, I consider this change involves lower risk than doing a bigger refactoring in order to be able to use `ExpectedArtifact` in the other cases.

Reply:
Method and property types should use an interface type, so these should change to `Map<String, Object>`

Reply:
@robzienert map type comment addressed